### ISSUE TITLE
Embed for allowed domain only

### ISSUE THREAD

**Issue by fmohican** [Chocobozzz/PeerTube#3241]
Hello,
Its possible to allow only those specified to embed video from selfhosted peertube?

For example:
peertube is : abdc.com
Domain that are allow to embed:
xyz.com
abc.com

Rest display an error, or simple blank page.

is that possible? how?

**Comment by rigelk** [comment]
I suppose you mean something like https://support.google.com/youtube/answer/6301625?hl=en, which we could implement.

In the meantime, if you are the administrator of your own instance, you can deny any any request on the embed location that doesn't have a referrer matching your domain.

**Comment by Findus23** [comment]
Couldn't one configure this simply by adding a `X-Frame-Options: allow-from https://example.com/` header?
https://developer.mozilla.org/de/docs/Web/HTTP/Headers/X-Frame-Options

**Comment by rigelk** [comment]
@Findus23 yes, a much better solution indeed!

**Comment by fmohican** [comment]
as fair as i know, that option with `X-Frame-Options: allow-from https://example.com/` aren't support by all browsers and can be bypassed.
But thank you.

**Comment by Findus23** [comment]
Hi,

You are right, I didn't know that. But the docs also say that you can use the CSP `frame-ancestors` directive for the same result (which should work in [most browsers](https://caniuse.com/mdn-http_headers_csp_content-security-policy_frame-ancestors))

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors

**Comment by fmohican** [comment]
Work like charm! I just set header in nginx and everything is okay.

**Comment by fmohican** [comment]
i will leave there instruction for nginx
```
add_header Content-Security-Policy "frame-ancestors domain.one domain.two sub.domain.two subsub.domain.one;"; 
```

**Comment by zookz** [comment]
> Content-Security-Policy

Where to add this in the nginx session?

**Comment by candidexmedia** [comment]
Restricting embeds to chosen domains is now available since [version 8.1](https://joinpeertube.org/news/release-8.1), on a per video basis. 🥳

<img width="1080" height="1358" alt="Image" src="https://github.com/user-attachments/assets/620ea5b9-08a5-45d4-b868-778f14aff6b7" />

I just wish it could be applied on a per channel and per account basis as well.
